plugins/outputs/postgresql/postgresql.go

```diff
@@ -31,50 +36,72 @@ func init() {
 
 func newPostgresql() *Postgresql {
 	return &Postgresql{
-		Schema:         "public",
-		TableTemplate:  "CREATE TABLE IF NOT EXISTS {TABLE}({COLUMNS})",
-		TagTableSuffix: "_tag",
+		Schema:                      "public",
+		TableTemplate:               "CREATE TABLE IF NOT EXISTS {TABLE}({COLUMNS})",
+		TagTableSuffix:              "_tag",
+		CachedTagsetsPerMeasurement: 1000,
+		DoSchemaUpdates:             true,
 	}
 }
 
 // Connect establishes a connection to the target database and prepares the cache
 func (p *Postgresql) Connect() error {
-	db, err := newDbWrapper(p.Address)
+	p.dbConnLock = &sync.Mutex{}
```

Comment:
Making `dbConnLock` a value instead of a pointer means you don't have to do this initialisation. Currently one goroutine could Connect and then call a method that takes the lock, but another goroutine could call Connect and replace the mutex.